pkg/uhttp/wrapper.go

```diff
@@ -279,6 +295,22 @@ func (c *BaseHttpClient) Do(req *http.Request, options ...DoOption) (*http.Respo
 	case http.StatusRequestTimeout:
 		return resp, status.Error(codes.DeadlineExceeded, resp.Status)
 	case http.StatusTooManyRequests:
+		retryAfter := resp.Header.Get("Retry-After")
```

Comment:
Ideally we'd make this code as general as possible, so that any 4xx/5xx status code with rate limit headers gets the rate limit description set on the error. Then the function signatures in syncer.go and connectorbuilder.go wouldn't need to change.

Reply:
I made that code into a function in recent commits, however, the only status errors that carry a `retry-after` header are `Too Many Requests`, `Service Unavailable`, and `Moved Permanently`; AFAIK.